feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/navigation/AccountSetupNavHost.kt

```diff
@@ -130,7 +130,17 @@ fun AccountSetupNavHost(
         composable(route = NESTED_NAVIGATION_SPECIAL_FOLDERS) {
             SpecialFoldersScreen(
                 onNext = {
-                    navController.navigate(NESTED_NAVIGATION_DISPLAY_OPTIONS)
+                    navController.navigate(NESTED_NAVIGATION_DISPLAY_OPTIONS) {
+                        if (hasSpecialFolders) {
```

Comment:
`hasSpecialFolders` is `true` when the special folders screen should be displayed. It doesn't indicate whether special folders have been configured automatically. (Maybe we need to change the name)

My suggestion is to add an additional callback parameter `onAutoAdvance` to `SpecialFoldersScreen` to be called when special folders have been found and the screen advanced automatically. 

That way we can pop `SpecialFoldersScreen` off the back stack when auto advancing (`onAutoAdvance`) and leave it on the back stack when the user had to configure special folders and manually pressed "Next" (`onNext`).